pkg/rbd/rbd_util_test.go

```diff
@@ -54,3 +54,68 @@ func TestHasSnapshotFeature(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateAndGetImageFeaturesWithMounter(t *testing.T) {
+	tests := []struct {
+		volOptions     map[string]string
+		isImageFeature string
+		isMounter      string
+		isErrMsg       string
+	}{
+		{
+			map[string]string{"imageFeatures": "layering", "mounter": rbdDefaultMounter},
+			"layering",
+			rbdDefaultMounter,
+			"",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-lock", "mounter": rbdNbdMounter},
+			"layering,exclusive-lock",
+			rbdNbdMounter,
+			"",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-lock,journaling", "mounter": rbdNbdMounter},
+			"layering,exclusive-lock,journaling",
+			rbdNbdMounter,
+			"",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,journaling", "mounter": rbdNbdMounter},
+			"",
+			"",
+			"feature journaling requires exclusive-lock",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-lock,journaling"},
+			"",
+			"",
+			"feature exclusive-lock requires rbd-nbd for mounter",
+		},
+		{
+			map[string]string{"imageFeatures": "layering,exclusive-lock,journaling", "mounter": rbdDefaultMounter},
+			"",
+			"",
+			"feature exclusive-lock requires rbd-nbd for mounter",
+		},
```

Comment:
Can you add a test for an image feature that does not exist? Imagine a typo in the `imageFeatures` value.

Reply:
I added a unit test for when the image feature does not exist.